lib/DxilContainer/DxcContainerBuilder.cpp

```diff
@@ -208,4 +200,20 @@ HRESULT DxcContainerBuilder::UpdateParts(AbstractMemoryStream *pStream) {
     if (cbWritten != pBlob->GetBufferSize()) { return E_FAIL; }
   }
   return S_OK;
-}
\ No newline at end of file
+}
+
+void DxcContainerBuilder::AddPart(DxilPart&& part) {
+  PartList::iterator it = std::find_if(m_parts.begin(), m_parts.end(), [&](DxilPart checkPart) {
+    return checkPart.m_fourCC == part.m_fourCC;
+  });
+  IFTBOOL(it == m_parts.end(), DXC_E_DUPLICATE_PART);
+  if (m_HasPrivateData) {
+    // Keep PrivateData at end, since it may have unaligned size.
+    m_parts.insert(&m_parts.back(), std::move(part));
```

Comment:
That's not standard C++, is it? Perhaps `.end() - 1` instead of `.back()`?

Reply:
Since m_parts is a SmallVector, iterator is actually just `typedef T *iterator;` where T is the element type, so an address to an element of the vector is fine as an iterator here.  But `end() - 1` is fine too of course.